packages/snaps-sdk/src/jsx/component.ts

```diff
@@ -0,0 +1,91 @@
+import type { Json } from '@metamask/utils';
+
+/**
+ * A key, which can be a string or a number.
+ */
+export type Key = string | number;
+
+/**
+ * A JSON object.
+ */
+export type JsonObject = Record<string, Json>;
+
+/**
+ * A generic JSX element.
+ *
+ * @property type - The type of the element.
+ * @property props - The props of the element.
+ * @property key - The key of the element.
+ */
+export type GenericSnapElement = {
+  type: string;
+  props: JsonObject;
+  key: Key | null;
+};
+
+/**
+ * A JSX element.
+ *
+ * @property type - The type of the element.
+ * @property props - The props of the element.
+ * @property key - The key of the element.
+ */
+export type SnapElement<
+  Props extends JsonObject = Record<string, never>,
```

Comment:
```suggestion
  Props extends JsonObject = Record<string, Json>,
```

Can this not be strictly typed?

Reply:
`never` is more strict than `Json` 😅 `Record<string, never>` is essentially an empty object. Note that it extends `JsonObject`, so it must be a JSON-compatible object, but we default to no props.